### PR TITLE
Added C1X Adapter

### PR DESCRIPTION
**Pull request by CathyHuangtw**
# Type of change
- [x] New bidder adapter  

## Description of change
- test parameters for validating bids
```
{
  bidder: 'c1x'
}
```

- contact email of the adapter’s maintainer: cathy@c1exchange.com
- [x] official adapter submission

## Note
- test ad unit
```
var adUnits = [
    {
         code: 'div-gpt-ad-1498193081995-0',
         sizes: [[300, 250], [300, 600]],
         bids: [
             {
                      bidder: 'c1x',
                      params: {
                          siteId: '999'
                      }
              }
         ]
    }
];
```

### CODE REVIEW

**Comment by CathyHuangtw** [comment]
Hi, would you be able to do a review on our adapter this week? Please let me know if you have further questions :) Thank you.

File: modules/c1xBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    BIDDER_CODE = 'c1x';
+
+  var pbjs = window.pbjs;
+
+  pbjs._c1xResponse = function(c1xResponse) {
+    if (c1xResponse) {
+      var response = c1xResponse;
+
+      if (typeof c1xResponse === CONSTANTS.objectType_string) {
+        response = JSON.parse(c1xResponse);
```
Comment (matthewlane) [inline modules/c1xBidAdapter.js]: Please wrap `JSON.parse` in a `try/catch` block to catch any parsing exceptions. See `appnexusAstBidAdapter` or other adapters for examples of this if needed

File: modules/c1xBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      }
+      for (var i = 0; i < response.length; i++) {
+        var data = response[i],
+          bidObject = null;
+        if (data.bid) {
+          bidObject = bidfactory.createBid(1);
```
Comment (matthewlane) [inline modules/c1xBidAdapter.js]: Prefer `CONSTANTS.STATUS.GOOD` in place of `1`

File: modules/c1xBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      var slots = pbjs.adUnits;
+      utils.logWarn(LOG_MSG.invalidBid);
+      for (var i = 0; i < slots.length; i++) {
+        bidmanager.addBidResponse(slots[i].code, noBidResponse());
+      }
+    }
+  };
+
+  function noBidResponse() {
+    var bidObject = bidfactory.createBid(2);
```
Comment (matthewlane) [inline modules/c1xBidAdapter.js]: Prefer `CONSTANTS.STATUS.NO_BID` in place of `2`

**Comment by CathyHuangtw** [comment]
@matthewlane Thank you for your review. I've also added a doc for our bidder to update on Prebid website in [this repo](https://github.com/prebid/prebid.github.io/pull/305). Will our adapter be merged soon and be included in your next release? Thanks!

File: test/spec/modules/c1xBidAdapter_spec.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    adId: 'div-c1x-ht',
+    cpm: 3.31,
+    ad: '<div><a target=\"_new\" href=\"http://c1exchange.com\"><img src=\"https://placeholdit.imgix.net/~text?txtsize=38&txt=C1X%20Ad%20300x250&w=300&h=250&txttrack=0\"></a></div>',
+    width: 300,
+    height: 250
+  };
+};
+
+describe('c1x adapter tests: ', () => {
+  let pbjs = window.pbjs || {};
```
Comment (dbemiller) [inline test/spec/modules/c1xBidAdapter_spec.js]: should use $$PREBID_GLOBAL$$ too.

**Comment by CathyHuangtw** [comment]
@dbemiller  thank you for taking the time to review my adapter code. I have fixed and pushed the changes. Hope it can get merged soon :)

File: test/spec/modules/c1xBidAdapter_spec.js (hunk 4 above)
Comment (CathyHuangtw) [inline test/spec/modules/c1xBidAdapter_spec.js]: thanks for the reminder. Have corrected this for the global naming convention.

**Pull request merged**